readme.md

```diff
@@ -45,6 +45,21 @@ Because this proposal allows existing contracts to remove higher-precision math
 ## Technical Specification
 
 The limit on the maximum length of Bitcoin Cash VM numbers (A.K.A. `nMaxNumSize`) is removed.
+The numbers will still be limited, but by the maximum stack item size (A.K.A. `MAX_SCRIPT_ELEMENT_SIZE`).
+The effective limits will then be a calculated value from `MAX_SCRIPT_ELEMENT_SIZE`:
+
+- `MAX_SCRIPTNUM = (MAX_SCRIPT_ELEMENT_SIZE * 8 - 1) - 1`, and
+- `MIN_SCRIPTNUM = -MAX_SCRIPTNUM`,
+
+rather than independently set limits (as they were before this upgrade).
+
+The [`CHIP: Targeted Virtual Machine Limits`](https://github.com/bitjson/bch-vm-limits) will set `MAX_SCRIPT_ELEMENT_SIZE` to 10,000 bytes, therefore valid range for results of arithmetic operations shall be the inclusive range: `[-2^79999 + 1, 2^79999 - 1]`.
+
+Any operation that would result in a stack item bigger than `MAX_SCRIPT_ELEMENT_SIZE` must fail, e.g. `{stack: a} OP_1ADD OP_0NOTEQUAL` must fail for `a = MAX_SCRIPTNUM`.
```

Comment:
`OP_0NOTEQUAL` makes it ambiguous whether it's enforced on input or output. I would just remove it.

Reply:
sry, it could still fail for a=-1 rather than overflow

must be `{stack: a} OP_1ADD OP_DROP OP_1` so if it fails, overflow can be the only reason

Reply:
But you already specified `a` ?

Reply:
scratch that yeah we can just say `{stack: a} OP_1ADD` must fail for a=MAX, we're not testing for pass in this example

Reply:
fixed https://github.com/bitjson/bch-bigint/pull/8/commits/2fe544b9673df58759fd0186b838f12f568b841b